xbmc/XBDateTime.cpp

```diff
@@ -799,123 +588,71 @@ bool CDateTime::SetDate(int year, int month, int day)
 
 bool CDateTime::SetTime(int hour, int minute, int second)
 {
-  // 01.01.1601 00:00:00 is 0 as filetime
-  return SetDateTime(1601, 1, 1, hour, minute, second);
-}
+  m_time = date::sys_seconds(std::chrono::seconds(0)) + std::chrono::hours(hour) +
+           std::chrono::minutes(minute) + std::chrono::seconds(second);
 
-void CDateTime::GetAsSystemTime(KODI::TIME::SystemTime& time) const
-{
-  FileTimeToSystemTime(&m_time, &time);
+  SetValid(true);
+  return true;
 }
 
-#define UNIX_BASE_TIME 116444736000000000LL /* nanoseconds since epoch */
 void CDateTime::GetAsTime(time_t& time) const
 {
-  long long ll = (static_cast<long long>(m_time.highDateTime) << 32) + m_time.lowDateTime;
-  time=(time_t)((ll - UNIX_BASE_TIME) / 10000000);
+  time = date::floor<std::chrono::seconds>(m_time.time_since_epoch())
+                .count();
 }
 
 void CDateTime::GetAsTm(tm& time) const
 {
-  KODI::TIME::SystemTime st;
-  GetAsSystemTime(st);
+  auto dp = date::floor<date::days>(m_time);
+
+  auto ymd = date::year_month_day{dp};
+  time.tm_year = int(ymd.year()) - 1900;
+  time.tm_mon  = unsigned(ymd.month()) - 1;
+  time.tm_mday = unsigned(ymd.day());
 
-  time.tm_year = st.year - 1900;
-  time.tm_mon = st.month - 1;
-  time.tm_wday = st.dayOfWeek;
-  time.tm_mday = st.day;
-  time.tm_hour = st.hour;
-  time.tm_min = st.minute;
-  time.tm_sec = st.second;
+  auto hms = date::make_time(date::floor<std::chrono::seconds>(m_time - dp));
+  time.tm_hour = hms.hours().count();
+  time.tm_min  = hms.minutes().count();
+  time.tm_sec  = hms.seconds().count();
 
-  mktime(&time);
+  date::weekday wd{dp};
+  time.tm_wday = wd.c_encoding();
```

Comment:
@HowardHinnant has said a possible alternate to work with 2.4.1 if decided minimum required for date lib is 2.4.1

```
time.tm_wday = (wd - date::Sunday) / date::days{1};
```